File: code/src/StandardBank.ConcessionManagement.Model/UserInterface/Glms/GlmsConcessionDetail.cs
using System.Collections.Generic;

namespace StandardBank.ConcessionManagement.Model.UserInterface.Glms
{
    /// <summary>
    /// Glms concession detail entity
    /// </summary>
    public class GlmsConcessionDetail : BaseConcessionDetail
    {
        public int GlmsConcessionDetailId { get; set; }
        public string LegalEntity { get; set; }
        public string GlmsProduct { get; set; }
        public int interestPricingCategoryId { get; set; }
        public int? GroupNumber { get; set; }
        public int? fkGlmsProductId { get; set; }    
        public int? fkLegalEntityAccountId { get; set; }
        public int? fkLegalEntityId { get; set; }
        public int InterestTypeId { get; set; }
        public int SlabTypeId { get; set; }
        public int GlmsGroupId { get; set; }
        public string InterestPricingCategory { get; set; }
 
        public IEnumerable<GlmsTierData> GlmsTierData { get; set; }

        public IEnumerable<GlmsTierDataView> GlmsTierDataView { get; set; }


    }
}

File: code/src/StandardBank.ConcessionManagement.UI/client-src/app/models/glms-concession-detail.ts
import { BaseConcessionDetail } from "./base-concession-detail";
import { GlmsTierData } from "./glms-tier-data";
import { GlmsTierDataView } from "./glms-tier-data-view";

export class GlmsConcessionDetail extends BaseConcessionDetail {

    glmsConcessionDetailId: number;
    legalEntity: string;   
    glmsProduct: string;
    glmsProductType: string;
    interestPricingCategoryId: number;
    glmsGroupId: number;
    groupNumber: number;
    interestTypeId: number;
    rateTypeId: number;
    slabTypeId: number;  
    tieredFrom: number;
    tieredTo: number;
    baseRate: number;
    value: number;
    glmsTierData: GlmsTierData[];
    glmsTierDataView: GlmsTierDataView[];
}
